[SVR-63] DB 분리 및 버전 관리 위해 Flyway 도입

sam971114 opened the pull request:
# 📌 개요
- DB를 aws rds로 분리 및 버전 관리를 위해 flyway를 도입했습니다.

# 💻 작업사항
- [x] RDS로 데이터베이스 분리 후, 기존 데이터 이전
- [x] Flyway 도입 및 DDL Auto 비활성화
- [x] 서버에 배포되어있는 docker-compose.yml 환경변수 수정

# ❌ 주의사항
- 현재 show,reservation은 각각 1개씩만 만들어둔 상태입니다. 참고 부탁드립니다.

# 💡 코드 리뷰 요청사항
-

sam971114 committed:
feat: admin cicd 구축

sam971114 committed:
feat: cicd 환경변수 수정

sam971114 committed:
feat: ec2 username 수정

sam971114 committed:
feat: docker-compose url 수정

sam971114 committed:
feat aws login 추가

sam971114 committed:
feat: docker image name 수정

sam971114 committed:
feat: docker registry 수정

sam971114 committed:
feat: 환경변수 변경

sam971114 committed:
feat: Jar파일 재생성

sam971114 committed:
feat: jar 파일 수정

sam971114 committed:
feat: gradle clean build 설정

sam971114 committed:
feat: test 제외

sam971114 committed:
feat: local docker 확인 완료

sam971114 committed:
feat: registry login 구현

sam971114 committed:
feat: env 추가

sam971114 committed:
[SVR-46] 관리자 티켓 예매 내역 조회 (#59)

* feat: update Ticket transaction 내용 추가

* feat: 전체 티켓 페이지별 조회 구현

* feat: 티켓 상태별로 조회 구현

* feat: 사용자 이름별로 티켓 조회 API 구현

* feat: 전화번호로 검색 API 구현 완료

* feat:공연 날짜 필터로 검색 API 구현 완료

* feat: 유저 타입별로 검색 API 구현 완료

* feat: 생성일자로 티켓 조회 API 구현 완료

* feat: 수정일자로 검색 필터 API 구현 완료

* feat: 티켓 검색 필터 구현

sam971114 committed:
hotfix: Pagination 구현 내용 수정 (#60)

* feat: 매개변수 @ModelAttribute로 수정

* feat: 연,월,일만 입력받을 수 있도록 변경

* feat: showDate 입력 형식 변경

* feat: page 중복 속성 삭제

0703kyj committed:
[SVR-49] refactor: 티켓 타입별 조회 로직 리팩토링 (#61)

* [SVR-49] 티켓조회 기능 리팩토링

* [SVR-49] 예외 수정

0703kyj committed:
[SVR-48] 실시간 입장 내역 확인 api 추가 (#62)

* [SVR-48] 실시간 입장조회 기능 추

* [SVR-48] api 트래픽 제어 구현

* [SVR-48] aop로 수정

* [SVR-48] 횟수 증가

0703kyj committed:
[SVR-50] 트랜젝션 처리 추 (#63)

sam971114 committed:
feat: Service 코드 lazy loading error fix (#66)

0703kyj committed:
Revert "feat: Service 코드 lazy loading error fix (#66)" (#67)

This reverts commit 46ed1a2b7e62970330664f191f3a3dbb74d30193.

sam971114 committed:
[SVR-51] 티켓 전화번호 수정 및 마스킹 처리 (#68)

* feat: 티켓 검색 전화번호 뒷자리 4자리 수정 구현

* feat: showDate 다음날까지 조회되는 내용 수정 구현

* feat: Masking 구현

* feat: ticketResponse 마스킹 구현

sam971114 committed:
feat: DB 이전 및 주소 변경

sam971114 committed:
feat: 초기 schema data 구현

sam971114 committed:
feat: flyway migration 작업전 repair 설정

sam971114 committed:
feat: admin redis 주소 변경

0703kyj commented on .github/workflows/ci-api.yml:
```diff
@@ -43,4 +43,4 @@ jobs:
 
       - name: Build with Gradle
-        run: ./gradlew clean build
+        run: ./gradlew clean build -x test
```
오잉 요거 테스트 왜 빠졌나요?

sam971114 replied:
빠졌던 test 다시 추가해두었고, 메인 모듈인 ticket 모듈에서만 flyway를 통해 DB를 관리하도록 수정했습니다!
추가적으로 merge 되어서 추가된 entity 내용들까지 V2 파일로 flyway 통해 추가해두었습니다.

0703kyj reviewed (commented):
admin이랑 ticket이랑 같은 db를 바라볼텐데 스크립트가 두벌로 관리되는 이유가 있나용??

sam971114 committed:
feat: flyway 의존성 메인 모듈에만 설정

sam971114 committed:
feat: flyway test 환경 세팅

sam971114 committed:
Merge branch 'dev' of https://github.com/DCNJ-Uket/Uket-BE into feature/SVR-54

sam971114 committed:
feat: dev branch merge

sam971114 committed:
feat: test code에 entity 추가

sam971114 committed:
Merge branch 'dev' of https://github.com/DCNJ-Uket/Uket-BE into feature/SVR-54

sam971114 committed:
feat: V3 schema 추가

sam971114 commented:
아래에 resolved된 대화에 답변을 달아놔 답변이 안보였던 것 같은데
빠졌던 test 다시 추가해두었고, 메인 모듈인 ticket 모듈에서만 flyway를 통해 DB를 관리하도록 수정했습니다!
추가적으로 merge 되어서 추가된 entity 내용들까지 V3 파일로 flyway 통해 추가해두었습니다.

0703kyj commented on application/admin-app-api/src/main/resources/application.yml:
```diff
@@ -120,12 +120,12 @@ spring:
   jpa:
     hibernate:
-      ddl-auto: update
+      ddl-auto: none
```
요거 validate 로 부탁드려용

0703kyj commented on application/ticket-app-api/src/main/resources/application.yml:
```diff
@@ -117,12 +117,18 @@ spring:
   jpa:
     hibernate:
-      ddl-auto: update
+      ddl-auto: none
```
요것도 마찬가지로 validate 부탁드립니다!

sam971114 committed:
Merge branch 'dev' of https://github.com/DCNJ-Uket/Uket-BE into feature/SVR-54

sam971114 committed:
feat: 유저 탈퇴시에 예매 reservedCount 감소 구현 (#88)

sam971114 committed:
Merge branch 'dev' of https://github.com/DCNJ-Uket/Uket-BE into feature/SVR-54

nonaninona committed:
[SVR-66] 티켓 조회에 지인 응답 추가 (#85)

* Update README.md

* Update README.md

* admin to develop (#64)

* feat: admin cicd 구축

* feat: cicd 환경변수 수정

* feat: ec2 username 수정

* feat: docker-compose url 수정

* feat aws login 추가

* feat: docker image name 수정

* feat: docker registry 수정

* feat: 환경변수 변경

* feat: Jar파일 재생성

* feat: jar 파일 수정

* feat: gradle clean build 설정

* feat: test 제외

* feat: local docker 확인 완료

* feat: registry login 구현

* feat: env 추가

* [SVR-46] 관리자 티켓 예매 내역 조회 (#59)

* feat: update Ticket transaction 내용 추가

* feat: 전체 티켓 페이지별 조회 구현

* feat: 티켓 상태별로 조회 구현

* feat: 사용자 이름별로 티켓 조회 API 구현

* feat: 전화번호로 검색 API 구현 완료

* feat:공연 날짜 필터로 검색 API 구현 완료

* feat: 유저 타입별로 검색 API 구현 완료

* feat: 생성일자로 티켓 조회 API 구현 완료

* feat: 수정일자로 검색 필터 API 구현 완료

* feat: 티켓 검색 필터 구현

* hotfix: Pagination 구현 내용 수정 (#60)

* feat: 매개변수 @ModelAttribute로 수정

* feat: 연,월,일만 입력받을 수 있도록 변경

* feat: showDate 입력 형식 변경

* feat: page 중복 속성 삭제

* [SVR-49] refactor: 티켓 타입별 조회 로직 리팩토링 (#61)

* [SVR-49] 티켓조회 기능 리팩토링

* [SVR-49] 예외 수정

* [SVR-48] 실시간 입장 내역 확인 api 추가 (#62)

* [SVR-48] 실시간 입장조회 기능 추

* [SVR-48] api 트래픽 제어 구현

* [SVR-48] aop로 수정

* [SVR-48] 횟수 증가

* [SVR-50] 트랜젝션 처리 추 (#63)

---------

Co-authored-by: sam971114 <simonana@naver.com>
Co-authored-by: sam971114 <56977023+sam971114@users.noreply.github.com>

* Revert "admin to develop (#64)" (#65)

This reverts commit 044e66d91f6dda8876ccf74e96874c865be492e3.

* [SVR-52] 입금 링크 제공 re-merge (#70)

* feat : Events 엔티티에 depositUrl 필드 추가

* feat : CheckTicketDto에 depositUrl 추가, "from" static factory method에 depositUrl set 구문 추가

* feat : TicketingResponse에 depositUrl 추가를 위해, TicketDto에 depositUrl 추가(builder set 구문에 추가)

* Revert "feat : TicketingResponse에 depositUrl 추가를 위해, TicketDto에 depositUrl 추가(builder set 구문에 추가)"

This reverts commit b1444d728dda27946d00c82ac5b1e988de76a9b2.

* Revert "feat : CheckTicketDto에 depositUrl 추가, "from" static factory method에 depositUrl set 구문 추가"

This reverts commit cb087b8bbad2791240c66163bf431229bd10fcdc.

* feat : TicketApi에 입금 링크 조회 api 명세 추가

* feat : TicketInfoService에서 ticketId를 가지고 depositUrl을 구하는 로직 작성

* feat : TicketController에서 getDepositUrl 메서드 구현

* test : 티켓에 대한 입금 계좌 링크를 찾는 테스트 케이스 추가

* feat : 티켓의 입금 계좌 링크를 구하는 로직 위치 변경 ticketInfoService(application) -> ticketService(domain)

* test : 로직 위치 변경에 따른 test 코드 수정

* feat : ticket 엔티티에 getDepositUrl 추가

디미터 법칙을 준수하여
다른 위치에서 티켓에 대한 입금 계좌 링크를 얻고자 할 때
event에 대한 의존을 없앨 수 있음

* feat : 입금 링크 정보 제공 중 티켓에 대한 유효성 검증 추가

입금 대기 중 상태가 아니면, 입금 링크를 제공하지 않음.

* feat : 입금 링크 정보 제공 중 티켓/유저에 대한 유효성 검증 추가

유저가 소유한 티켓이 맞는 지 확인함.

* test : 로직 이동으로 인한 TicketInfoServiceTest에서 티켓에_대한_입금_계좌_링크를_찾는다 테스트 케이스 삭제

* feat : Ticket의 getDepositUrl 메서드 제거

변경 가능성이 낮은 부분에 대한 캡슐화 제거

* hotfix: depositUrl lazy loading error fix (#72)

* feat: db url 수정

* feat: lazyLoading 관련 내용 수정

* [SVR-56] 입금 링크 제공 API 수정 - 명세 및 예외처리 추가 및 계좌 정보 제공 (#74)

* feat : DepositUrl이 존재하지 않는 경우 예외처리 추가

* feat : getDepositUrl에 대한 명세 수정

400 - TI0014
404 - TI0009, TI0015

에 대한 명세로 올바르게 수정

* feat : Events에 accountDetail 필드 추가

* feat : Events에 accountNumber, accountOwner 필드 추가

* feat : getDepositUrl -> getAccountInfo / String -> AccountInfoResponse

* feat : TicketService의 getDepositUrl -> getAccountInfo

* feat : TicketController의 getAccountInfo 구현

* feat : TicketController의 getAccountInfo 명세에서 TI0015 케이스 제거

* feat : getAccountInfo API 명세 @Operation 내용 수정

* feat : Account 엔티티 추가

* feat : AccountRepository, AccountService 추가

* feat : account api 추가

* feat : 기존 구현 제거

* feat : @Transactional read 추가

* feat : getAccount 명세 오류 케이스 수정

* feat : 티켓팅 완료 시, 유저가 티켓 정보 조회 시, eventId를 함께 넘겨주도록 변경

* feat : 티켓팅 완료 시, 유저가 티켓 정보 조회 시, eventId를 함께 넘겨주도록 변경

* [SVR-57] 베너 RedirectUrl 추가 (#75)

* [SVR-61] 계좌정보 lazy loading 문제 수정 (#79)

* [SVR-61] 티켓금액 추가

* feat: 회원탈퇴 기능 구현 (#78)

* [SVR-53] 질의응답 도메인 작성 (#71)

* feat : 질의응답 도메인 모듈 생성(form-domain)

* feat : FormDto, SurveyDto 추가

* feat : FormDto, SurveyDto 추가

* feat : Survey 정보 제공 기능 추가

* feat : 응답 검증 및 저장 기능 추가

* feat : FormException이 BaseException을 상속하도록 변경

* feat : Form에서 answer 삭제

* feat : submitResponse 협력 과정 수정

String response의 map 형태가 아닌 Answer List를 받도록 함
Survey는 해당 answer를 적절한 form에게 전달
form은 해당 answer를 validatation 함

* feat : DropdownForm, DropdownAnswer 추가

* feat : Form, Answer -> TextForm, TextAnswer로 변경

* feat : Answer, Form 인터페이스 추가

* feat : Survey를 Answer, Form 인터페이스에 맞춰 작성

* feat : Survey를 Answer, Form 인터페이스에 맞춰 작성

* feat : validateAnswer에 타입 체크 추가

* feat : FormDto -> TextFormDto로 이름 변경

* feat : SurveyDto 추가, FormDtoMapper 이용

* feat : SurveyDtoTest 추가

* feat : repository 타입 교정

* feat : FormType 추가

* feat : FormType 추가

* hotfix: depositUrl lazy loading error fix (#72)

* feat: db url 수정

* feat: lazyLoading 관련 내용 수정

* feat : 정보 제공 (SurveyDto를 통한) 기능 추가

* feat : 유저 응답에 대한 Answer 생성 및 검증, 저장 기능 추가

* style : Survey의 handleUserReponse -> createAnswers로 이름 변경

* feat : Answer를 JPA 엔티티로 등록, Id를 generated value로 설정 후 생성자에서 제거

* style : 불필요 import 문 삭제

* feat : Answer에 userId 추가

* feat : Answer 엔티티의 userId, formId 필드 -> Users user, Form form으로 변경(JPA)

* feat : Form을 JPA 엔티티로 변경

* feat : Survey를 JPA 엔티티로 변경 + form과의 관계 추가

* feat : repository들을 JPA repository로 변경

* feat : survey에서 event로 many to one 관계 추가

* feat : form 도메인에 user/event에 대한 의존성 추가

* feat : Survey와 Events 간의 관계를 1대1로 수정. Event를 주인으로 설정

* fix : Answer form에 대한 joinColumn이 잘못된 문제 수정

* file : Answer, Form 구현 클래스 삭제

* file : UserResponseDto -> FormResponseDto 이름 변경

* feat : answer 생성 책임 survey, form -> service

* feat : Option 엔티티 추가

* feat : Option 엔티티와 Form 엔티티를 N대1로 연결

* feat : Answer에서 타입에 따른 검증 로직 추가

* feat : submitResponse에 @Transactional 추가

* file : FormDto 상속 관계 파일 삭제

* feat : OptionDto 추가 및 FormDto, SurveyDto 수정

* feat : Option 불필요 메서드 제거

* feat : Answer가 form이 가진 정보에 대해선 묻는 방식으로 validation 하도록 변경

* style : 불필요 import문 제거

* test : Dropdown 응답에 대한 테스트, Text 응답에 대한 테스트 추가

* test : FormService에 대한 테스트 추가

* fix : Event와 Survey 매핑 구문에 문제가 있던 점 수정

* chore : event-domain이 form-domain에 의존하도록 수정

* fix : mappedBy 구문 오류 수정

* fix : Option이 MySQL 예약어인 관계로 Options로 수정

* feat : FormException을 위한 ErroCode 추가 및 적용

* [SVR-56] 입금 링크 제공 API 수정 - 명세 및 예외처리 추가 및 계좌 정보 제공 (#74)

* feat : DepositUrl이 존재하지 않는 경우 예외처리 추가

* feat : getDepositUrl에 대한 명세 수정

400 - TI0014
404 - TI0009, TI0015

에 대한 명세로 올바르게 수정

* feat : Events에 accountDetail 필드 추가

* feat : Events에 accountNumber, accountOwner 필드 추가

* feat : getDepositUrl -> getAccountInfo / String -> AccountInfoResponse

* feat : TicketService의 getDepositUrl -> getAccountInfo

* feat : TicketController의 getAccountInfo 구현

* feat : TicketController의 getAccountInfo 명세에서 TI0015 케이스 제거

* feat : getAccountInfo API 명세 @Operation 내용 수정

* feat : Account 엔티티 추가

* feat : AccountRepository, AccountService 추가

* feat : account api 추가

* feat : 기존 구현 제거

* feat : @Transactional read 추가

* feat : getAccount 명세 오류 케이스 수정

* feat : 티켓팅 완료 시, 유저가 티켓 정보 조회 시, eventId를 함께 넘겨주도록 변경

* feat : 티켓팅 완료 시, 유저가 티켓 정보 조회 시, eventId를 함께 넘겨주도록 변경

* [SVR-57] 베너 RedirectUrl 추가 (#75)

* [SVR-61] 계좌정보 lazy loading 문제 수정 (#79)

* [SVR-61] 티켓금액 추가

* feat: 회원탈퇴 기능 구현 (#78)

* feat: event에 해당하는 survey 조회 API 구현

* feat: survey 답변 작성 API 구현

* feat: test 코드 수정

* feat: swagger 예외 추가

* feat: Lazy loading 오류 fix

* feat: 불필요 event repository 로직 삭제

* feat: Survey 조회 swagger 예외 추가

---------

Co-authored-by: sam971114 <56977023+sam971114@users.noreply.github.com>
Co-authored-by: Youngjun Kim <127181370+0703kyj@users.noreply.github.com>
Co-authored-by: 0703kyj <kyj0703@konkuk.ac.kr>
Co-authored-by: sam971114 <simonana@naver.com>

* feat : searchAllTickets에서 answer 정보까지 불러오도록 수정

* feat : TicketSearchService로 로직 이동

* feat : @Transactional 추가

* feat : TicketingResponse 추가 및 searchTickets 수정

* feat : ApplyMasking에 TicketingResponse의 마스킹 방식 등록

* feat : ApplyMasking의 typeValue에 TicketingResponse 등록

* feat : 티켓별 질의응답 결과를 FormAnswerDto를 통해 전달하도록 수정

* test : TicketSearchServiceTest 추가

* style : 주석 todo로 변경

---------

Co-authored-by: sam971114 <56977023+sam971114@users.noreply.github.com>
Co-authored-by: Youngjun Kim <127181370+0703kyj@users.noreply.github.com>
Co-authored-by: sam971114 <simonana@naver.com>
Co-authored-by: 0703kyj <kyj0703@konkuk.ac.kr>

sam971114 committed:
Merge branch 'dev' of https://github.com/DCNJ-Uket/Uket-BE into feature/SVR-54

sam971114 committed:
feat: ddl-auto validate로 변경

nonaninona committed:
fix : [SVR-72] answer를 찾을 수 없는 경우에 대한 예외처리 (#90)

* Update README.md

* Update README.md

* admin to develop (#64)

* feat: admin cicd 구축

* feat: cicd 환경변수 수정

* feat: ec2 username 수정

* feat: docker-compose url 수정

* feat aws login 추가

* feat: docker image name 수정

* feat: docker registry 수정

* feat: 환경변수 변경

* feat: Jar파일 재생성

* feat: jar 파일 수정

* feat: gradle clean build 설정

* feat: test 제외

* feat: local docker 확인 완료

* feat: registry login 구현

* feat: env 추가

* [SVR-46] 관리자 티켓 예매 내역 조회 (#59)

* feat: update Ticket transaction 내용 추가

* feat: 전체 티켓 페이지별 조회 구현

* feat: 티켓 상태별로 조회 구현

* feat: 사용자 이름별로 티켓 조회 API 구현

* feat: 전화번호로 검색 API 구현 완료

* feat:공연 날짜 필터로 검색 API 구현 완료

* feat: 유저 타입별로 검색 API 구현 완료

* feat: 생성일자로 티켓 조회 API 구현 완료

* feat: 수정일자로 검색 필터 API 구현 완료

* feat: 티켓 검색 필터 구현

* hotfix: Pagination 구현 내용 수정 (#60)

* feat: 매개변수 @ModelAttribute로 수정

* feat: 연,월,일만 입력받을 수 있도록 변경

* feat: showDate 입력 형식 변경

* feat: page 중복 속성 삭제

* [SVR-49] refactor: 티켓 타입별 조회 로직 리팩토링 (#61)

* [SVR-49] 티켓조회 기능 리팩토링

* [SVR-49] 예외 수정

* [SVR-48] 실시간 입장 내역 확인 api 추가 (#62)

* [SVR-48] 실시간 입장조회 기능 추

* [SVR-48] api 트래픽 제어 구현

* [SVR-48] aop로 수정

* [SVR-48] 횟수 증가

* [SVR-50] 트랜젝션 처리 추 (#63)

---------

Co-authored-by: sam971114 <simonana@naver.com>
Co-authored-by: sam971114 <56977023+sam971114@users.noreply.github.com>

* Revert "admin to develop (#64)" (#65)

This reverts commit 044e66d91f6dda8876ccf74e96874c865be492e3.

* [SVR-52] 입금 링크 제공 re-merge (#70)

* feat : Events 엔티티에 depositUrl 필드 추가

* feat : CheckTicketDto에 depositUrl 추가, "from" static factory method에 depositUrl set 구문 추가

* feat : TicketingResponse에 depositUrl 추가를 위해, TicketDto에 depositUrl 추가(builder set 구문에 추가)

* Revert "feat : TicketingResponse에 depositUrl 추가를 위해, TicketDto에 depositUrl 추가(builder set 구문에 추가)"

This reverts commit b1444d728dda27946d00c82ac5b1e988de76a9b2.

* Revert "feat : CheckTicketDto에 depositUrl 추가, "from" static factory method에 depositUrl set 구문 추가"

This reverts commit cb087b8bbad2791240c66163bf431229bd10fcdc.

* feat : TicketApi에 입금 링크 조회 api 명세 추가

* feat : TicketInfoService에서 ticketId를 가지고 depositUrl을 구하는 로직 작성

* feat : TicketController에서 getDepositUrl 메서드 구현

* test : 티켓에 대한 입금 계좌 링크를 찾는 테스트 케이스 추가

* feat : 티켓의 입금 계좌 링크를 구하는 로직 위치 변경 ticketInfoService(application) -> ticketService(domain)

* test : 로직 위치 변경에 따른 test 코드 수정

* feat : ticket 엔티티에 getDepositUrl 추가

디미터 법칙을 준수하여
다른 위치에서 티켓에 대한 입금 계좌 링크를 얻고자 할 때
event에 대한 의존을 없앨 수 있음

* feat : 입금 링크 정보 제공 중 티켓에 대한 유효성 검증 추가

입금 대기 중 상태가 아니면, 입금 링크를 제공하지 않음.

* feat : 입금 링크 정보 제공 중 티켓/유저에 대한 유효성 검증 추가

유저가 소유한 티켓이 맞는 지 확인함.

* test : 로직 이동으로 인한 TicketInfoServiceTest에서 티켓에_대한_입금_계좌_링크를_찾는다 테스트 케이스 삭제

* feat : Ticket의 getDepositUrl 메서드 제거

변경 가능성이 낮은 부분에 대한 캡슐화 제거

* hotfix: depositUrl lazy loading error fix (#72)

* feat: db url 수정

* feat: lazyLoading 관련 내용 수정

* [SVR-56] 입금 링크 제공 API 수정 - 명세 및 예외처리 추가 및 계좌 정보 제공 (#74)

* feat : DepositUrl이 존재하지 않는 경우 예외처리 추가

* feat : getDepositUrl에 대한 명세 수정

400 - TI0014
404 - TI0009, TI0015

에 대한 명세로 올바르게 수정

* feat : Events에 accountDetail 필드 추가

* feat : Events에 accountNumber, accountOwner 필드 추가

* feat : getDepositUrl -> getAccountInfo / String -> AccountInfoResponse

* feat : TicketService의 getDepositUrl -> getAccountInfo

* feat : TicketController의 getAccountInfo 구현

* feat : TicketController의 getAccountInfo 명세에서 TI0015 케이스 제거

* feat : getAccountInfo API 명세 @Operation 내용 수정

* feat : Account 엔티티 추가

* feat : AccountRepository, AccountService 추가

* feat : account api 추가

* feat : 기존 구현 제거

* feat : @Transactional read 추가

* feat : getAccount 명세 오류 케이스 수정

* feat : 티켓팅 완료 시, 유저가 티켓 정보 조회 시, eventId를 함께 넘겨주도록 변경

* feat : 티켓팅 완료 시, 유저가 티켓 정보 조회 시, eventId를 함께 넘겨주도록 변경

* [SVR-57] 베너 RedirectUrl 추가 (#75)

* [SVR-61] 계좌정보 lazy loading 문제 수정 (#79)

* [SVR-61] 티켓금액 추가

* feat: 회원탈퇴 기능 구현 (#78)

* [SVR-53] 질의응답 도메인 작성 (#71)

* feat : 질의응답 도메인 모듈 생성(form-domain)

* feat : FormDto, SurveyDto 추가

* feat : FormDto, SurveyDto 추가

* feat : Survey 정보 제공 기능 추가

* feat : 응답 검증 및 저장 기능 추가

* feat : FormException이 BaseException을 상속하도록 변경

* feat : Form에서 answer 삭제

* feat : submitResponse 협력 과정 수정

String response의 map 형태가 아닌 Answer List를 받도록 함
Survey는 해당 answer를 적절한 form에게 전달
form은 해당 answer를 validatation 함

* feat : DropdownForm, DropdownAnswer 추가

* feat : Form, Answer -> TextForm, TextAnswer로 변경

* feat : Answer, Form 인터페이스 추가

* feat : Survey를 Answer, Form 인터페이스에 맞춰 작성

* feat : Survey를 Answer, Form 인터페이스에 맞춰 작성

* feat : validateAnswer에 타입 체크 추가

* feat : FormDto -> TextFormDto로 이름 변경

* feat : SurveyDto 추가, FormDtoMapper 이용

* feat : SurveyDtoTest 추가

* feat : repository 타입 교정

* feat : FormType 추가

* feat : FormType 추가

* hotfix: depositUrl lazy loading error fix (#72)

* feat: db url 수정

* feat: lazyLoading 관련 내용 수정

* feat : 정보 제공 (SurveyDto를 통한) 기능 추가

* feat : 유저 응답에 대한 Answer 생성 및 검증, 저장 기능 추가

* style : Survey의 handleUserReponse -> createAnswers로 이름 변경

* feat : Answer를 JPA 엔티티로 등록, Id를 generated value로 설정 후 생성자에서 제거

* style : 불필요 import 문 삭제

* feat : Answer에 userId 추가

* feat : Answer 엔티티의 userId, formId 필드 -> Users user, Form form으로 변경(JPA)

* feat : Form을 JPA 엔티티로 변경

* feat : Survey를 JPA 엔티티로 변경 + form과의 관계 추가

* feat : repository들을 JPA repository로 변경

* feat : survey에서 event로 many to one 관계 추가

* feat : form 도메인에 user/event에 대한 의존성 추가

* feat : Survey와 Events 간의 관계를 1대1로 수정. Event를 주인으로 설정

* fix : Answer form에 대한 joinColumn이 잘못된 문제 수정

* file : Answer, Form 구현 클래스 삭제

* file : UserResponseDto -> FormResponseDto 이름 변경

* feat : answer 생성 책임 survey, form -> service

* feat : Option 엔티티 추가

* feat : Option 엔티티와 Form 엔티티를 N대1로 연결

* feat : Answer에서 타입에 따른 검증 로직 추가

* feat : submitResponse에 @Transactional 추가

* file : FormDto 상속 관계 파일 삭제

* feat : OptionDto 추가 및 FormDto, SurveyDto 수정

* feat : Option 불필요 메서드 제거

* feat : Answer가 form이 가진 정보에 대해선 묻는 방식으로 validation 하도록 변경

* style : 불필요 import문 제거

* test : Dropdown 응답에 대한 테스트, Text 응답에 대한 테스트 추가

* test : FormService에 대한 테스트 추가

* fix : Event와 Survey 매핑 구문에 문제가 있던 점 수정

* chore : event-domain이 form-domain에 의존하도록 수정

* fix : mappedBy 구문 오류 수정

* fix : Option이 MySQL 예약어인 관계로 Options로 수정

* feat : FormException을 위한 ErroCode 추가 및 적용

* [SVR-56] 입금 링크 제공 API 수정 - 명세 및 예외처리 추가 및 계좌 정보 제공 (#74)

* feat : DepositUrl이 존재하지 않는 경우 예외처리 추가

* feat : getDepositUrl에 대한 명세 수정

400 - TI0014
404 - TI0009, TI0015

에 대한 명세로 올바르게 수정

* feat : Events에 accountDetail 필드 추가

* feat : Events에 accountNumber, accountOwner 필드 추가

* feat : getDepositUrl -> getAccountInfo / String -> AccountInfoResponse

* feat : TicketService의 getDepositUrl -> getAccountInfo

* feat : TicketController의 getAccountInfo 구현

* feat : TicketController의 getAccountInfo 명세에서 TI0015 케이스 제거

* feat : getAccountInfo API 명세 @Operation 내용 수정

* feat : Account 엔티티 추가

* feat : AccountRepository, AccountService 추가

* feat : account api 추가

* feat : 기존 구현 제거

* feat : @Transactional read 추가

* feat : getAccount 명세 오류 케이스 수정

* feat : 티켓팅 완료 시, 유저가 티켓 정보 조회 시, eventId를 함께 넘겨주도록 변경

* feat : 티켓팅 완료 시, 유저가 티켓 정보 조회 시, eventId를 함께 넘겨주도록 변경

* [SVR-57] 베너 RedirectUrl 추가 (#75)

* [SVR-61] 계좌정보 lazy loading 문제 수정 (#79)

* [SVR-61] 티켓금액 추가

* feat: 회원탈퇴 기능 구현 (#78)

* feat: event에 해당하는 survey 조회 API 구현

* feat: survey 답변 작성 API 구현

* feat: test 코드 수정

* feat: swagger 예외 추가

* feat: Lazy loading 오류 fix

* feat: 불필요 event repository 로직 삭제

* feat: Survey 조회 swagger 예외 추가

---------

Co-authored-by: sam971114 <56977023+sam971114@users.noreply.github.com>
Co-authored-by: Youngjun Kim <127181370+0703kyj@users.noreply.github.com>
Co-authored-by: 0703kyj <kyj0703@konkuk.ac.kr>
Co-authored-by: sam971114 <simonana@naver.com>

* [SVR-62]  유저 탈퇴시 예매 가능 티켓 수 증가 (#80)

* feat: 유저 탈퇴시 기존 티켓의 예매 가능 수 증가

* feat: reservation id 받도록 수정

* feat: ticket 조회시 reservation까지 한번에 가지고오도록 수정

* feat: surveyResponse에서 중복된 응답 항목 제거 (#84)

* [SVR-64] 티켓 기간 만료 및 입장 완료 예외 처리 (#81)

* feat: 티켓 취소시 취소,기간 만료 티켓 예외처리

* feat: user ticket 조회 시 만료된 티켓 예외처리

* feat: ticket cancel validation시에 입장 완료된 ticket 예외 처리

* feat: Ticket 조회 로직 변경에 따른 test 코드 로직 변경

* feat: repository query 수정

* [SVR-65] 약관 조회 API 추가 (#82)

* [SVR-65] 약관 도메인 세팅

* [SVR-65] 약관 조회 API 추가

* [SVR-65] BaseEntity 추가

* [SVR-65] 코드오너 추가

* [SVR-65] User 의존성 제거

* [SVR-65] 약관 동의 API 추가 (#83)

* [SVR-65] 약관 동의 API 추기

* [SVR-65] API Operation 수정

* [SVR-58] 공연 목록 조회 시 시작, 종료 일자 반환하도록 수정 (#76)

* [SVR-58] 공연 목록 조회 시 시작, 종료 일자 반환하도록 수정

* [SVR-58] 테스트 수정

* [SVR-58] 시작시간 반환하도록 수정

* [SVR-58] 시작시간 찾는 부분 메서드 분리

* [SVR-58] 테스트 수정

* [SVR-68] 약관 동의 API 에러 응답 문서화 (#86)

* hotfix: user 탈퇴시 유저 answer 삭제 (#87)

* hotfix: 카카오 로그인 시, 닉네임 활용하도록 수정

* hotfix: 약관 API 회원가입 예외 처리

* feat: form이 필수적인지 아닌지 boolean 변수를 추가했습니다. (#89)

* feat : findAnswerByFormIdAndUserId가 null을 반환했을 때 예외처리 추가

* feat : merge conflict 해결

* feat : findAnswerByFormIdAndUserId에서 Answer에 대한 예외 처리(Answer 존재여부, 필수 응답 질문)

* test : 테스트 반영

* test : 필수 응답에 대한 예외처리 추가

---------

Co-authored-by: sam971114 <56977023+sam971114@users.noreply.github.com>
Co-authored-by: Youngjun Kim <127181370+0703kyj@users.noreply.github.com>
Co-authored-by: sam971114 <simonana@naver.com>
Co-authored-by: 0703kyj <kyj0703@konkuk.ac.kr>

nonaninona committed:
fix : [SVR-74] 예매 목록 조회 시 예외처리 수정 (#94)

* fix : Answer에 대한 예외처리를 예외 throw -> 빈 Answer 반환으로 수정

* feat : noAnswerDto의 response 값을 "" -> "미응답"으로 변경

nonaninona committed:
fix : AnswerDto에 response 값을 전달하지 않던 문제 수정

sam971114 committed:
feat: 초기 schema enum을 varchar로 변경

sam971114 committed:
feat: shcema v2 account 추가 및 deposit_url 삭제

sam971114 committed:
feat: 추가 domain 관련 sql 구문 생성

sam971114 committed:
feat: auto-conflict merge

sam971114 committed:
feat: test schema 작성

sam971114 committed:
feat: merge conflict fix

sam971114 committed:
feat: flyway db enum column varchar로 수정

sam971114 committed:
feat: admin과 도메인 충돌 dto 분리

sam971114 committed:
feat: flyway test code 추가

sam971114 committed:
feat: duplicate test code delete
